runtime-modules/content/src/lib.rs

```diff
@@ -1717,37 +1864,47 @@ decl_module! {
             origin,
             owner_id: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
             video_id: T::VideoId,
+            winner_id: T::MemberId,
+            commit: CurrencyOf<T>, // amount the auctioner is committed to
         ) {
-
-            // Ensure given video exists
-            let video = Self::ensure_video_exists(&video_id)?;
+            let winner_account_id = T::MemberAuthenticator::controller_account_id(winner_id)?;
 
             // Ensure nft is already issued
-            let nft = video.ensure_nft_is_issued::<T>()?;
+            let nft = Self::ensure_nft_exists(video_id)?;
+            let video = Self::video_by_id(video_id);
+
+            // Ensure auction for given video id exists, retrieve corresponding one
+            let auction = Self::ensure_open_auction_state(&nft)?;
 
             // Ensure actor is authorized to accept open auction bid
             ensure_actor_authorized_to_manage_nft::<T>(origin, &owner_id, &nft.owner, video.in_channel)?;
 
-            // Ensure auction for given video id exists, retrieve corresponding one
-            let auction = nft.ensure_auction_state::<T>()?;
-
-            // Ensure open type auction
-            auction.ensure_is_open_auction::<T>()?;
+            // Ensure open auction bid exists
+            let bid = Self::ensure_open_bid_exists(video_id, winner_id)?;
 
-            // Ensure there is a bid to accept
-            let bid = auction.ensure_last_bid_exists::<T>()?;
+            // Ensure bid is related to ongoing auction
+            bid.ensure_bid_is_relevant::<T>(auction.auction_id)?;
 
-            let owner_account_id = Self::ensure_owner_account_id(&video, &nft).ok();
+            // Ensure commit matches amount
+            bid.ensure_valid_bid_commit::<T>(commit)?;
 
             //
             // == MUTATION SAFE ==
             //
 
-            let nft = Self::complete_auction(video.in_channel, nft, bid, owner_account_id);
-            let video = video.set_nft_status(nft);
+            let post_auction_nft = Self::complete_auction(
+                &nft,
+                video.in_channel,
+                winner_account_id,
+                winner_id,
+                bid.amount,
+            );
+
+            // remove bid
+            OpenAuctionBidByVideoAndMember::<T>::remove(video_id, winner_id);
 
             // Update the video
-            VideoById::<T>::insert(video_id, video);
+            VideoById::<T>::mutate(video_id, |v| v.set_nft_status(post_auction_nft));
 
             // Trigger event
             Self::deposit_event(RawEvent::OpenAuctionBidAccepted(owner_id, video_id));
```

Comment:
This now needs to identify which bid was accepted

Reply:
:+1:

Reply:
fixed in 4aa92e0d73f2e9ba4d534d9aa695df51bc901781